### ISSUE TITLE
Import include/exclude csv files for segmentation

### ISSUE THREAD

**Issue by amurphy-cl** [CarnegieLearningWeb/UpGrade#530]
**Is your feature request related to a problem? Please describe.**
Currently the UpGrade UI allows for include/exclude members to be added on a by-individual basis or via a json file. Since we often have large lists (in csv format) of members, we need an import function to more easily add list members in a format that most users will have on-hand. 

**Describe the solution you'd like**
The ability to upload a csv file, perhaps with fixed headers, in the Segments tab for either include or exclude lists.

**Comment by ppratikcr7** [comment]
@zackcl Do we need to have a frontend UI for user's to download a CSV format which they can feed in their data?

**Comment by zackcl** [comment]
@ppratikcr7 I'm not sure if I understood this. Do you mean a UI to select a format (JSON or CSV) when the user clicks the "Export" button on the Segment details page?

<img width="750" alt="Screen Shot 2022-09-19 at 9 46 45 AM" src="https://user-images.githubusercontent.com/90279765/191032225-8dd8c48a-54f1-4e9e-a1b0-eb818717b2a2.png">

**Comment by ppratikcr7** [comment]
No, actually the one you mentioned should already be there. But, we should also provide a button/link "Download Import Format" to let users download a sample csv format in which they can feed the large list of members as per the fixed header format we provide them and later they can import this csv with member data.

**Comment by zackcl** [comment]
OK, so I guess that button should be on the Segments detail page? Or how about adding a link to the "Import Segment" modal something like the following?

<img width="750" alt="Screen Shot 2022-09-19 at 11 53 13 AM" src="https://user-images.githubusercontent.com/90279765/191060024-3556fdea-610b-434a-829a-d0f3abb617c6.png">

**Comment by ppratikcr7** [comment]
I feel this simple link to download is a good option. Let's take views from other's as well.

**Comment by amurphy-cl** [comment]
raising the priority of this as it's going to be used frequently for upcoming experiments

**Comment by ppratikcr7** [comment]
> raising the priority of this as it's going to be used frequently for upcoming experiments

@amurphy-cl So are we fine with the above design as suggested by Zack. I feel it's good for now. Just wanted to confirm before we start development on this.

**Comment by amurphy-cl** [comment]
Looks great to me - thanks

**Comment by RidhamShah** [comment]
@amurphy-cl is there any format decided for CSV of Segments? We can have columns of individuals, groups, and subSegments but we still need `context` information. We also need ID/name which I think can be managed through CSV title.

This is a sample CSV format that we can look at and update - 
[segmnetCSVFormat.csv](https://github.com/CarnegieLearningWeb/UpGrade/files/12881883/segmnetCSVFormat.csv)

**Comment by amurphy-cl** [comment]
Per meeting discussion today, let's use this format. Thanks! 
[segmentCSVFormat_new.csv](https://github.com/CarnegieLearningWeb/UpGrade/files/12882907/segmentCSVFormat_new.csv)

**Comment by Yagnik56** [comment]
@amurphy-cl what will be the format of export document, will we add fields like id, description, segment type in csv file ?

**Comment by Yagnik56** [comment]
@amurphy-cl this issue doesn't mention about export of segment csv so should I create separate issue and PR for that?

**Comment by amurphy-cl** [comment]
@danoswaltCL yeah please create a separate issue for segment export

**Comment by amurphy-cl** [comment]
What do we think about this format for import/export? 
[segmentCSV_Import-Export_Format.csv](https://github.com/CarnegieLearningWeb/UpGrade/files/13501484/segmentCSV_Import-Export_Format.csv)

**Comment by Yagnik56** [comment]
Hey @amurphy-cl,
I am adding the recording here for importing segment csv, have a look and let me know if you wish to change anything.
https://github.com/CarnegieLearningWeb/UpGrade/assets/50392803/fb77012c-4e3e-4371-96ed-d67c34b49304

**Comment by amurphy-cl** [comment]
The main thing I'm not sure I like is that there is that the Type field encompasses both app context as well as segment member type (individual, schoolid, etc). I get that this makes the import/export simpler but it doesn't seem as intuitive or user-friendly. @SritterCL any thoughts?

**Comment by zackcl** [comment]
@Yagnik56 I apologize for suggesting this design change after you've already started implementing CSV upload in the Import modal, but I believe the CSV file should only contain the list of members (Type and ID pairs) and not the entire segment, which includes additional information such as name, description, and app context. This additional information should be included in the JSON file. Therefore I suggest the following design (the image shows the modal to add/edit a segment): 

![Experiments - export pop up-05 (2) (1)](https://github.com/CarnegieLearningWeb/UpGrade/assets/90279765/a0c4db86-9639-4dee-8e80-d9522c0549cd)

Here are the details on how it should work: Clicking the 'Import CSV' button allows the user to choose a CSV file from their device. Once the file is selected, it replaces the existing rows with the imported members in the table. While not ideal, I believe it's a suitable solution for now.

**Comment by zackcl** [comment]
QA: @zackcl
